linera-storage/src/db_storage.rs

```diff
@@ -575,7 +576,7 @@ where
         limit: u32,
     ) -> Result<Vec<HashedCertificateValue>, ViewError> {
         let mut hash = Some(from);
-        let mut values = Vec::new();
+        let mut values = Vec::with_capacity(limit as usize);
```

Comment:
This loop has `break`s in it, and often won't actually yield that many elements.